dqc_us_rules/dqc_us_0041.py

```diff
@@ -0,0 +1,345 @@
+_RULE_VERSION = '1.0'
+
+
+ugtDocs = (
+    {"year": 2012,
+     "namespace": "http://fasb.org/us-gaap/2012-01-31",
+     "docLB": "http://xbrl.fasb.org/us-gaap/2012/elts/us-gaap-doc-2012-01-31.xml",  # noqa
+     "entryXsd": "http://xbrl.fasb.org/us-gaap/2012/entire/us-gaap-entryPoint-std-2012-01-31.xsd",  # noqa
+     },
+
+    {"year": 2013,
+     "namespace": "http://fasb.org/us-gaap/2013-01-31",
+     "docLB": "http://xbrl.fasb.org/us-gaap/2013/elts/us-gaap-doc-2013-01-31.xml",  # noqa
+     "entryXsd": "http://xbrl.fasb.org/us-gaap/2013/entire/us-gaap-entryPoint-std-2013-01-31.xsd",  # noqa
+     },
+
+    {"year": 2014,
+     "namespace": "http://fasb.org/us-gaap/2014-01-31",
+     "docLB": "http://xbrl.fasb.org/us-gaap/2014/elts/us-gaap-doc-2014-01-31.xml",  # noqa
+     "entryXsd": "http://xbrl.fasb.org/us-gaap/2014/entire/us-gaap-entryPoint-std-2014-01-31.xsd",  # noqa
+     },
+
+    {"year": 2015,
+     "namespace": "http://fasb.org/us-gaap/2015-01-31",
+     "docLB": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/elts/us-gaap-doc-2015-01-31.xml",  # noqa
+     "entryXsd": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/entire/us-gaap-entryPoint-std-2015-01-31.xsd",  # noqa
+     },
+
+    {"year": 2016,
+     "namespace": "http://fasb.org/us-gaap/2016-01-31",
+     "docLB": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/elts/us-gaap-doc-2016-01-31.xml",  # noqa
+     "entryXsd": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/entire/us-gaap-entryPoint-std-2016-01-31.xsd",  # noqa
+     }
+)
+
+
+def _make_cache(
+        val, ugt, cntlr, ugt_calcs_json_file, ugt_default_dimensions_json_file
+):
+    """
+    Creates a new caches for the Taxonomy default dimensions
+
+    :param val: ValidateXbrl to be validated
+    :type val: :class: '~arelle.ValidateXbrl.ValidateXbrl'
+    :param ugt: Taxonomy to check
+    :type ugt: str
+    :param cntlr: cntlr to save to
+    :type cntlr: :class: '~arelle.CntrlWinMain'
+    :param ugt_calcs_json_file: where to save json taxonomies
+    :type ugt_calcs_json_file: str
+    :param ugt_default_dimensions_json_file: location to save json default
+        dimensions
+    :type ugt_default_dimensions_json_file: str
+    :return: no explicit return, but saves caches for dqc_us_0041
+    :rtype: None
+    """
+    val.modelXbrl.modelManager.addToLog(
+                    _("loading us-gaap {0} calculations and default "  # noqa
+                      "dimensions into cache").format(ugt["year"])
+                )
+    started_at = time.time()
+    ugt_entry_xsd = ugt["entryXsd"]
+    val.usgaapCalculations = {}
+    val.usgaapDefaultDimensions = {}
+    prior_validate_disclosure_system = (
+        val.modelXbrl.modelManager.validateDisclosureSystem
+    )
+    val.modelXbrl.modelManager.validateDisclosureSystem = False
+    calculations_instance = (ModelXbrl.load(
```

Comment:
I need this to get the default dimensions, but I don't have to save it so I have removed that part